torch/_inductor/dependencies.py

```diff
@@ -356,7 +347,8 @@ def extract_read_writes(
     prefix: str = "d",
 ):
     args, var_ranges = index_vars_squeeze(*argsizes, prefix=prefix)
-    rw = RecordLoadStore(var_ranges, normalize=normalize)
+    inner = _RecordLoadStoreInner(var_ranges=var_ranges, normalize=normalize)
```

Comment:
`RecordLoadStoreInner` inherits from `V.MockHandler` which also does string formatting but builds up everything in a single line with no use of varables, so the expressions can build up exponentially. e.g.
```python
b = a * a
c = b * b
d = c * c
```
will be represented as the string
`ops.mul(ops.mul(ops.mul(a, a), ops.mul(a, a)), ops.mul(a, a), ops.mul(a, a))), ops.mul(ops.mul(a, a), ops.mul(a, a)), ops.mul(a, a), ops.mul(a, a)))`


A better solution would be to have another base that returns `tmp{n}` without actually building up a string at all.

Reply:
I think that RecordLoadStoreInner doesn't even look at the returned outputs at all, and I was planning on getting rid of the MockHandler inheritance, so I guess I'll do it all in one go.

Reply:
I believe the only cases where it matters is when the return type isn't directly an IR value, so `index_expr` must return a sympy expression and `reduction` must sometimes return a tuple of IR values. What exactly that IR value is represented as doesn't matter.